File: experimental/data_pipeline_go/globi/globi.py
from pathlib import Path
from datetime import datetime, timezone
import requests


BASE_PATH = Path("/", "Users", "danielbrickner", "data")
URL = "https://zenodo.org/records/18220953/files/SInAS_Workflow_Manual_vs2.0.pdf?download=1"
SOURCE = "globi"
FILENAME = "SInAS_Workflow_Manual_vs2.0.pdf"


def get_file(path: Path) -> None:
    response = requests.get(URL)

    with open(path, 'wb') as f:
        f.write(response.content)

    print(f"Wrote: {path}")


def get_path(base_path: Path) -> Path:
    dt = datetime.now(tz=timezone.utc)
    dt_str = dt.isoformat()
    return base_path.joinpath(SOURCE, dt_str, FILENAME)


def make_path(path: Path) -> None:
    path.mkdir(parents=True, exist_ok=True)


def main():
    path = get_path(BASE_PATH)
    print(path)
    make_path(path.parent)
    get_file(path)


if __name__ == "__main__":
    main()

File: experimental/data_pipeline_go/globi/globi.go
package main

import (
	"fmt"
	"io"
	"net/http"
	"os"
	"time"

	"path/filepath"
)

// const url string = "https://zenodo.org/record/20546682/files/interactions.tsv.gz"
const url string = "https://zenodo.org/records/18220953/files/SInAS_3.1.1.csv?download=1"
const source = "globi"
const fileName = "SInAS_3.1.1.csv"
const newFileDirMode = 0775

func getBaseDirectory() string {
	return filepath.Join("/", "Users", "danielbrickner", "data")
}

func getFile(url string, path string) {
	out, err := os.Create(path)
	if err != nil {
		panic("dis ting broke mon")
	}
	defer out.Close()

	resp, err := http.Get(url)
	if err != nil {
		panic("Could not download")
	}
	defer resp.Body.Close()

	if resp.StatusCode == http.StatusOK {
		bodyBytes, err := io.ReadAll(resp.Body)
		if err != nil {
			panic(err)
		}
		os.WriteFile(path, bodyBytes, newFileDirMode)
	}
}

func getParentDownloadPath(basePath string) string {
	t := time.Now().UTC()
	dt := t.Format(time.RFC3339)
	return filepath.Join(basePath, source, dt)
}

func getDownloadPath(parentPath string) string {
	return filepath.Join(parentPath, fileName)
}

func makeParentDirectory(parentPath string) {
	os.MkdirAll(parentPath, newFileDirMode)
}

func blah() {
	baseDirectory := getBaseDirectory()
	parentDirectory := getParentDownloadPath(baseDirectory)
	makeParentDirectory(parentDirectory)
	downloadPath := getDownloadPath(parentDirectory)
	fmt.Println(downloadPath)
	getFile(url, downloadPath)
}
